fastly/ngwaf/v1/workspaces/api_create.go

```diff
@@ -22,16 +22,16 @@ type CreateInput struct {
 	// AttackSignalThresholds are the parameters for system site alerts.
 	AttackSignalThresholds *AttackSignalThresholdsCreateInput `json:"attack_signal_thresholds,omitempty"`
 	// ClientIPHeaders lists the request headers containing the client IP address.
-	ClientIPHeaders []string `json:"client_ip_headers"`
+	ClientIPHeaders []string `json:"client_ip_headers,omitempty"`
 	// Context, if supplied, will be used as the Request's context.
 	Context *context.Context `json:"-"`
 	// DefaultBlockingResponseCode is the default response code.
-	DefaultBlockingResponseCode *int `json:"default_blocking_response_code"`
+	DefaultBlockingResponseCode *int `json:"default_blocking_response_code,omitempty"`
 	// Description is the description of a workspace.
-	Description *string `json:"description"`
+	Description *string `json:"description,omitempty"`
 	// IPAnonymization is the selected option to anonymize IP addresses.
-	IPAnonymization *string `json:"ip_anonymization"`
-	// Mode is the mode of a workspace.
+	IPAnonymization *string `json:"ip_anonymization,omitempty"`
+	// Mode is the mode of a workspace (required).
```

Comment:
If `Mode` is actually required, then the `Create` function should ensure that it is provided.

Reply:
yes, `mode` is required and I missed the check.